src/hoc/RealTime.tsx

```diff
@@ -0,0 +1,53 @@
+'use client'
+
+import { supabase } from '@/lib/supabase'
+import { selectTaskBoard, setTasks } from '@/redux/features/taskBoardSlice'
+import store from '@/redux/store'
+import { TaskResponse } from '@/types/dto/tasks.dto'
+import { RealtimePostgresChangesPayload } from '@supabase/supabase-js'
+import { usePathname } from 'next/navigation'
+import { ReactNode, useEffect } from 'react'
+import { useSelector } from 'react-redux'
+import { useRouter } from 'next/navigation'
+
+interface RealTimeTaskResponse extends TaskResponse {
+  deletedAt: string
+}
+
+export const RealTime = ({ children }: { children: ReactNode }) => {
+  const { tasks, token } = useSelector(selectTaskBoard)
+  const pathname = usePathname()
+  const router = useRouter()
+
+  const handleTaskRealTimeUpdates = (payload: RealtimePostgresChangesPayload<RealTimeTaskResponse>) => {
+    if (payload.eventType === 'INSERT') {
+      store.dispatch(setTasks([...tasks, payload.new]))
+    }
+    if (payload.eventType === 'UPDATE') {
+      const updatedTask = payload.new
+      if (updatedTask.deletedAt) {
+        const newTaskArr = tasks.filter((el) => el.id !== updatedTask.id)
+        store.dispatch(setTasks(newTaskArr))
+        if (pathname.includes('detail')) {
+          router.push(`/?token=${token}`)
```

Comment:
can you explain why we are doing this?  adding a code comment here would be helpful

Reply:
We are doing this to redirect user to the `/` route if they are in `/details` route when the task is deleted.